### PR TITLE
[SPARK-16881][MESOS] Migrate Mesos configs to use ConfigEntry

### PR DESCRIPTION
**Pull request by techaddict**
## What changes were proposed in this pull request?

Migrate Mesos configs to use ConfigEntry
## How was this patch tested?

Jenkins Tests

### CODE REVIEW

**Comment by techaddict** [comment]
cc: @mgummelt @srowen

File: mesos/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala [hunk 1]
```diff
@@ -51,7 +52,7 @@ private[mesos] class MesosClusterDispatcher(
   extends Logging {
 
   private val publicAddress = Option(conf.getenv("SPARK_PUBLIC_DNS")).getOrElse(args.host)
-  private val recoveryMode = conf.get("spark.deploy.recoveryMode", "NONE").toUpperCase()
+  private val recoveryMode = conf.get(RECOVERY_MODE).getOrElse("NONE").toUpperCase()
```
Comment (mgummelt) [inline mesos/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala]: Shouldn't the "NONE" default be added to the config builder?

**Comment by mgummelt** [comment]
Thanks!  One small fix then LGTM

**Review by mgummelt — CHANGES_REQUESTED** [review]

**Comment by techaddict** [comment]
@mgummelt  done! 👍

**Comment by mgummelt** [comment]
cc @srowen for merge into master

**Comment by srowen** [comment]
Merged to master

**Comment by mgummelt** [comment]
Thanks @techaddict.  Another PR to move the rest of the configs would be very welcome.

**Comment by techaddict** [comment]
@mgummelt yes working on it.

**Pull request closed without merging**